WalletWasabi/Services/UpdateManager.cs

```diff
@@ -0,0 +1,253 @@
+using Newtonsoft.Json.Linq;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+using WalletWasabi.Logging;
+using WalletWasabi.Microservices;
+using WalletWasabi.Models;
+using WalletWasabi.Tor.Http;
+
+namespace WalletWasabi.Services;
+
+public class UpdateManager
+{
+	private string InstallerName { get; set; } = "";
+
+	public UpdateManager(string dataDir, IHttpClient httpClient)
+	{
+		DataDir = dataDir;
+		DownloadsDir = Path.Combine(DataDir, "Downloads");
+		HttpClient = httpClient;
+	}
+
+	public async void UpdateStatusChangedAsync(UpdateStatus updateStatus)
+	{
+		bool updateAvailable = !updateStatus.ClientUpToDate || !updateStatus.BackendCompatible;
+		Version targetVersion = updateStatus.ClientVersion;
+		Exception? downloadException = null;
+		if (!updateAvailable)
+		{
+			return;
+		}
+
+		byte retries = 0;
```

Comment:
`byte`?

Reply:
For smaller memory usage as tries is a small number(?)